### PR TITLE
ObjectCreator.create retains description summary note (abstract)

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Part of #782 - Argo collection registration takes a collection abstract, and we've been dropping it on the floor.  This allows dsa to retain it.

## How was this change tested?

added tests to test suite.

## Which documentation and/or configurations were updated?

na

### CODE REVIEW

File: app/services/cocina/mapper.rb [hunk 1]
```diff
@@ -131,9 +131,17 @@ def build_identification
...
 
+    # rubocop:disable Rails/DynamicFindBy
     def build_descriptive
-      { title: [{ status: 'primary', value: TitleMapper.build(item) }] }
+      desc = { title: [{ status: 'primary', value: TitleMapper.build(item) }] }
+
+      # collections are registered with abstracts
+      return desc if item.descMetadata.find_by_terms(:abstract).blank?
+
+      desc[:note] = [{ type: 'summary', value: item.descMetadata.find_by_terms(:abstract).text }]
```
Comment (ndushay) [inline app/services/cocina/mapper.rb:141]: this is an Om datastream, so we want `find_by_terms`, not the thing rubocop is asking for.

File: app/services/content_metadata_generator.rb [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # frozen_string_literal: true
 
 # Builds the contentMetadata xml from cocina filesets
-# rubocop:disable Metrics/ClassLength
```
Comment (ndushay) [inline app/services/content_metadata_generator.rb:4]: this was flagged by rubocop as no longer needed.

File: app/services/cocina/mapper.rb [hunk 3]
```diff
@@ -131,9 +131,17 @@ def build_identification
       end
     end
 
+    # rubocop:disable Rails/DynamicFindBy
     def build_descriptive
-      { title: [{ status: 'primary', value: TitleMapper.build(item) }] }
+      desc = { title: [{ status: 'primary', value: TitleMapper.build(item) }] }
+
+      # collections are registered with abstracts
+      return desc if item.descMetadata.find_by_terms(:abstract).blank?
```
Comment (jcoyne) [inline app/services/cocina/mapper.rb:139]: I find this easier to read if we avoid the low-level api (e.g. find_by_terms):

```suggestion
      return desc if item.descMetadata.abstract.blank?
```

File: app/services/cocina/mapper.rb (hunk 1 above)
Comment (jcoyne) [inline app/services/cocina/mapper.rb:141]:
```suggestion
      desc[:note] = [{ type: 'summary', value: item.descMetadata.abstract.first }]
```

Comment (ndushay) [reply app/services/cocina/mapper.rb]: I'd swear I had it this way and the tests didn't pass ... in any case I'm glad to make this change and have it work.

**Comment by ndushay** [comment]
@jcoyne thanks for the suggestion - I started out that way but must've done something wrong.  Glad for that fix!

**Pull request merged**